libyul/optimiser/Semantics.h

```diff
@@ -54,11 +54,50 @@ class SideEffectsCollector: public ASTWalker
 		Block const& _ast,
 		std::map<YulString, SideEffects> const* _functionSideEffects = nullptr
 	);
+	SideEffectsCollector(
+		Dialect const& _dialect,
+		ForLoop const& _ast,
+		std::map<YulString, SideEffects> const* _functionSideEffects = nullptr
+	);
 
 	using ASTWalker::operator();
 	void operator()(FunctionCall const& _functionCall) override;
 
 	bool movable() const { return m_sideEffects.movable; }
+
+	bool movableRelativeTo(SideEffects const& _other, bool _codeContainsMSize = true)
```

Comment:
It might be better to not have a default value here.

Reply:
Removed the default.